hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -2532,8 +2532,9 @@ public boolean processReport(final DatanodeID nodeID,
         return !node.hasStaleStorages();
       }
       if (context != null) {
-        if (!blockReportLeaseManager.checkLease(node, startTime,
-              context.getLeaseId())) {
+        if (!namesystem.isInStartupSafeMode()
```

Comment:
I think this is a dup of HDFS-12914. In HDFS-12914, this part of code is moved to another place to fix the bug.

This fix is essentially saying if namenode is in safe mode, don't check for lease.